nixos/modules/services/web-apps/zipline.nix

```diff
@@ -0,0 +1,112 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.zipline;
+in
+{
+  meta.maintainers = with lib.maintainers; [ defelo ];
+
+  options.services.zipline = {
+    enable = lib.mkEnableOption "Zipline";
+
+    package = lib.mkPackageOption pkgs "zipline" { };
+
+    settings = lib.mkOption {
+      type = lib.types.attrsOf lib.types.str;
+      description = ''
+        Configuration of Zipline. See <https://zipline.diced.sh/docs/config> for more information.
+      '';
+      default = { };
+      example = {
+        CORE_SECRET = "changethis";
+        CORE_DATABASE_URL = "postgres://postgres:postgres@postgres/postgres";
+        CORE_HOST = "0.0.0.0";
+        CORE_PORT = "3000";
+        DATASOURCE_LOCAL_DIRECTORY = "/var/lib/zipline/uploads";
+      };
+    };
+
+    environmentFiles = lib.mkOption {
+      type = lib.types.listOf lib.types.path;
+      default = [ ];
+      example = [ "/run/secrets/zipline.env" ];
+      description = ''
+        Files to load environment variables from (in addition to [](#opt-services.zipline.settings). This is useful to avoid putting secrets into the nix store. See <https://zipline.diced.sh/docs/config> for more information.
+      '';
+    };
+
+    database.createLocally = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = ''
+        Whether to create a local database with PostgreSQL.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    services.zipline.settings = {
+      CORE_DATABASE_URL = lib.mkIf cfg.database.createLocally "postgresql://zipline@localhost/zipline?host=/run/postgresql";
+      DATASOURCE_LOCAL_DIRECTORY = lib.mkDefault "/var/lib/zipline/uploads";
```

Comment:
That directory should be created with tmpfilesd

Reply:
Zipline automatically creates this directory if it doesn't exist yet

Reply:
Doesn't it need more permissions than necessary to do that?

Reply:
I don't think so. `/var/lib/zipline` is already created by systemd and zipline is the owner of this directory